internal/cli/command_factory.go

```diff
@@ -160,7 +162,20 @@ func (factory *CommandFactory) Run(cmd *cobra.Command) {
 			factory.errLogger.Fatal(err)
 		}
 
-		if printErr := factory.ui.Print(terminal.NewErrorLog(err)); printErr != nil {
+		logs := []terminal.Log{
+			terminal.NewErrorLog(err),
+		}
+
+		if e, ok := err.(realm.ErrInvalidSession); ok {
```

Comment:
so we actually don't care at this point if it's an `ErrInvalidSession` or not, we'll allow any type of error here define suggested commands or reference links.  what you're looking for here is something along the lines of:

```go
if e, ok := err.(CommandSuggester); ok {
  logs = append(logs, terminal.NewSuggestedCommandsLog(e.SuggestedCommands()))
}
if e, ok := err.(LinkReferrer); ok {
  logs = append(logs, terminal.NewReferenceLinksLog(e. ReferenceLinks()))
}
```